modules/react/pagination/lib/Pagination/usePaginationModel.ts

```diff
@@ -5,10 +5,29 @@ import {getRangeMax, getRangeMin} from './common/utils/helpers';
 import {PaginationModel} from './types';
 
 export type UsePaginationModelConfig = {
+  /**
+   * The page number for the last page (it can also be used as a total page
+   * count)
+   */
   lastPage: number;
+  /**
+   * The page number for the first page
+   * @default 1
+   */
   firstPage?: number;
```

Comment:
Maybe I'm missing something, but `firstPage` didn't work as I expected it to. For example, setting it to `2` still showed Page 1 in the range, although I couldn't interact with the Page 1 PageButton. How is it meant to work?

Reply:
😕 
Your observation matches implementation. In the `StepToPreviousButton` has an `isDisabled` to prevent going to a page less than the `firstPage` and the `goTo` event prevents going to a page lower than `firstPage`. The question might be why would a developer choose a "first page" other than 1

Reply:
Good find. There's a mismatch between what's here and what's in `buildPageRange` [here](https://github.com/Workday/canvas-kit/blob/master/modules/react/pagination/lib/Pagination/buildPageRange.ts#L16)

Reply:
> _The question might be why would a developer choose a "first page" other than 1_

That's fair. I thought about that a lot while building the component, but it felt like an artificial constraint. The underlying logic really doesn't care about "pages" (other than the bug you found). It's takes the range size, the current item, the first and last pages, and creates a list. Outside of that, it doesn't really care.

This is a range of -10 to 10
![](http://g.recordit.co/hT8n1MqDpU.gif)

Reply:
Would you ever use this? Probably not. But the component is sort of agnostic to that.